tensorflow_addons/rnn/cell.py

```diff
@@ -366,3 +366,425 @@ def _create_norm_layer(self, name):
+    ```
+    """
+
+    def __init__(
+            self,
+            units,
+            activation='tanh',
+            use_bias=True,
+            layernorm_epsilon=1e-05,  # NEW(!)
+            kernel_initializer='glorot_uniform',
+            recurrent_initializer='orthogonal',
+            bias_initializer='zeros',
+            gamma_initializer='ones',  # NEW(!)
+            kernel_regularizer=None,
+            recurrent_regularizer=None,
+            bias_regularizer=None,
+            gamma_regularizer=None,  # NEW(!)
+            kernel_constraint=None,
+            recurrent_constraint=None,
+            bias_constraint=None,
+            gamma_constraint=None,  # NEW(!)
+            dropout=0.,
+            recurrent_dropout=0.,
+            **kwargs):
+        super(LayerNormSimpleRNNCell, self).__init__(
+            units,
+            activation=activation,
+            use_bias=use_bias,
+            kernel_initializer=kernel_initializer,
+            recurrent_initializer=recurrent_initializer,
+            bias_initializer=bias_initializer,
+            kernel_regularizer=kernel_regularizer,
+            recurrent_regularizer=recurrent_regularizer,
+            bias_regularizer=bias_regularizer,
+            kernel_constraint=kernel_constraint,
+            recurrent_constraint=recurrent_constraint,
+            bias_constraint=bias_constraint,
+            dropout=dropout,
+            recurrent_dropout=recurrent_dropout,
+            dtype=kwargs.get('dtype'),
+            trainable=kwargs.get('trainable', True))
+        self.layernorm = keras.layers.LayerNormalization(
+            axis=-1,
+            epsilon=layernorm_epsilon,
+            center=False,
+            scale=True,
+            beta_initializer=None,
+            gamma_initializer=gamma_initializer,
+            beta_regularizer=None,
+            gamma_regularizer=gamma_regularizer,
+            beta_constraint=None,
+            gamma_constraint=gamma_constraint,
+            dtype=kwargs.get('dtype'),
+            trainable=kwargs.get('trainable', True))
+
+    def build(self, input_shape):
+        super(LayerNormSimpleRNNCell, self).build(input_shape)
+        self.layernorm.build((None, self.units))
+
+    def call(self, inputs, states, training=None):
+        """Formulas.
+
+        Notation:
+            y_t : Cell output at t (`output`)
+            y_{t-1} : Previous cell output at t-1 (`prev_output`)
+            x_t : The new input at t (`inputs`)
+            W_xh : Weight matrix for inputs x_t (`self.kernel`)
+            W_hh : Weights for prev. outputs y_{t-1} (`self.recurrent_kernel`)
+            b : Bias term for centering (`self.bias`)
+            d1 : Dropout function for x_t (`inputs * dp_mask`)
+            d2 : Dropout function for y_{t-1} (`prev_output * rec_dp_mask`)
+            ln : Scaling function from layer normalization (`self.layernorm`)
+            f : Activation function (`self.activation`)
+
+        Case 1:
```

Comment:
I think the comment need some updates now, since all the case should have layernorm.

Reply:
Case 1+3 refers to Keras' SimpleRNN, Case 2+4 refers to addons' LayerNormSimpleRNNCell. I made it more clear: Case 1, this is Keras SimpleRNN. Case 2, that's what is different in LayerNormSimpleRNNCell (Same for Case 3 -> 4).